napari/layers/points/points.py

```diff
@@ -1127,6 +1124,72 @@ def mode(self, mode):
 
         self.events.mode(mode=mode)
 
+    @property
+    def view_data(self):
+        """Get the coords of the points in view
+
+        Returns
+        -------
+        view_data : (N x D) np.ndarray
+            Array of coordinates for the N points in view
+        """
+        if len(self._indices_view) > 0:
+
+            data = self.data[np.ix_(self._indices_view, self.dims.displayed)]
+
+        else:
+            # if no points in this slice send dummy data
+            data = np.zeros((0, self.dims.ndisplay))
+
+        return data
+
+    @property
+    def view_size(self):
+        """Get the sizes of the points in view
+
+       Returns
+       -------
+       view_size : (N x D) np.ndarray
+           Array of sizes for the N points in view
+        """
+        if len(self._indices_view) > 0:
+            # Get the point sizes and scale for ndim display
+            sizes = (
+                self.size[
+                    np.ix_(self._indices_view, self.dims.displayed)
+                ].mean(axis=1)
+                * self._point_scale_view
+            )
+
+        else:
+            # if no points in this slice send dummy data
+            sizes = [0]
```

Comment:
I suspect it's because of some weird stuff inside the corresponding `vispy_points_layer.py` code https://github.com/napari/napari/blob/master/napari/_vispy/vispy_points_layer.py#L65 as vispy can't take an empty list there, but I'd instead make this return an empty list and then do that catch and conversion inside `vispy_points_layer.py`

Reply:
@sofroniewn is right. The `Points._view_size` now returns an empty list when there are no points in view.